x-pack/legacy/plugins/siem/public/pages/overview/events_by_dataset/index.tsx

```diff
@@ -75,26 +69,11 @@ const EventsByDatasetComponent: React.FC<Props> = ({
   const kibana = useKibana();
```

Comment:
GH won't let me comment on line `67` above, but it appears that `deleteQuery` should be declared as a dependency in:

```
  useEffect(() => {
    return () => {
      if (deleteQuery) {
        deleteQuery({ id: ID });
      }
    };
  }, []);
```